pkg/file/pipeline/writer.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package pipeline
+
+import "io"
+
+// ChainWriter is a writer in a pipeline.
+// It is up to the implementer to decide whether a writer
+// calls the next writer or not. Implementers should
+// call the Sum method of the subsequent writer in case there
+// exists one.
+type ChainWriter interface {
+	ChainWrite(*pipeWriteArgs) (int, error)
```

Comment:
All ChainWriter implementations return 0 int and ChainWrite calls do not check for this return value. Is it really needed?